src/main/scala/org/kidsfirstdrc/dwh/external/ImportClinVar.scala

```diff
@@ -1,47 +1,129 @@
+      .withColumn("clndn", split(concat_ws("", col("clndn")), "\\|"))
+      .withColumn("conditions", split(regexp_replace(concat_ws("|", col("clndn")), "_", " "), "\\|"))
+      .withColumn("clndisdbincl", split(concat_ws("", col("clndisdbincl")), "\\|"))
+      .withColumn("clndnincl", split(concat_ws("", col("clndnincl")), "\\|"))
+      .withColumn("mc", split(concat_ws("", col("mc")), "\\|"))
+      .withColumn("allele_origin", allele_origin_udf(col("origin")))
+      .drop("clin_sig_original")
+
+  }
+
+  override def load(data: DataFrame, output: String)(implicit spark: SparkSession): Unit = {
+    data.coalesce(1)
+      .write
+      .mode("overwrite")
+      .format("parquet")
+      .option("path", s"$output/clinvar")
+      .saveAsTable("variant.clinvar")
+  }
+
   def info_fields(df: DataFrame, excludes: String*): Seq[Column] = {
     df.columns.collect { case c if c.startsWith("INFO") && !excludes.contains(c) => col(c) as c.replace("INFO_", "").toLowerCase }
   }
 
-  val df = vcf(input)
-  df.select(chromosome +:
-    start +:
-    end +:
-    name +:
-    reference +:
-    alternate +:
-    ($"INFO_CLNSIG" as "clin_sig_original") +:
-    (split(regexp_replace($"INFO_CLNSIGCONF"(0), """\(.\)""", ""), "%3B") as "clin_sig_conflict") +:
-    info_fields(df, "INFO_CLNSIG", "INFO_CLNSIGCONF"): _*
-  )
-    .withColumn("clin_sig",
-      when(
-        array_contains($"clin_sig_original", "Conflicting_interpretations_of_pathogenicity"),
-        array_union(array_remove($"clin_sig_original", "Conflicting_interpretations_of_pathogenicity"), $"clin_sig_conflict")
-      ).otherwise($"clin_sig_original")
+  def info_fields_names(df: DataFrame, excludes: String*): Seq[String] = {
+    df.columns.collect { case c if c.startsWith("INFO") && !excludes.contains(c) => c.replace("INFO_", "").toLowerCase }
+  }
+
+  def allele_origin_udf: UserDefinedFunction = udf { array: mutable.WrappedArray[String] =>
+
+    val unknown = "unknown"
+
+    val labels = Map(
+      0 -> unknown,
+      1 -> "germline",
+      2 -> "somatic",
+      4 -> "inherited",
+      8 -> "paternal",
+      16 -> "maternal",
+      32 -> "de-novo",
+      64 -> "biparental",
+      128 -> "uniparental",
+      256 -> "not-tested",
+      512 -> "tested-inconclusive",
+      1073741824 -> "other"
     )
-    .coalesce(1)
-    .write
-    .mode("overwrite")
-    .format("parquet")
-    .option("path", s"$output/clinvar")
-    .saveAsTable("variant.clinvar")
+
+    array match {
+      //in case the inpout column is null return an Array("unknown")
+      case null =>
+        val wa: mutable.WrappedArray[String] = mutable.WrappedArray.make(Array(unknown))
+        wa
+      //in any other case transform the array of number into their labels
+      case _ =>
+        array
+          .map(_.toInt)
+          .flatMap {number =>
+            labels.collect {
+              case (_, _) if number == 0 => unknown
+              case (digit, str) if (number & digit) != 0 => str
```

Comment:
is it needed, in the second case, to test "number" for not equaling to zero since it s been tested in the previous case?